### PR TITLE
VB-615 Visit Outcomes

### PR DESCRIPTION
**Pull request by AledWEvans**
## What does this pull request do?

Add's visit outcomes to the visit notes

## What is the intent behind these changes?

allow legacy visit outcomes to be added to the new system

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt [hunk 1]
```diff
@@ -554,10 +573,41 @@ class VisitResourceIntTest : IntegrationTestBase() {
...
+        .jsonPath("$.visitNotes.length()").isEqualTo(updateRequest.visitNotes!!.size)
+        .jsonPath("$.visitNotes[0].type").isEqualTo(updateRequest.visitNotes!![0].type.name)
+        .jsonPath("$.visitNotes[1].type").isEqualTo(updateRequest.visitNotes!![1].type.name)
         .jsonPath("$.sessionId").isEqualTo(updateRequest.sessionId!!)
     }
 
+    @Test
+    fun `put visit by reference - delete notes`() {
+
+      val updateRequest = UpdateVisitRequest(
```
Comment (philip-milne) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:585]: This needs to be explicit on which not is being deleted / removed

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt [hunk 2]
```diff
@@ -554,10 +573,41 @@ class VisitResourceIntTest : IntegrationTestBase() {
         .jsonPath("$.visitorSupport.length()").isEqualTo(updateRequest.supportList!!.size)
         .jsonPath("$.visitorSupport[0].supportName").isEqualTo(updateRequest.supportList!![0].supportName)
         .jsonPath("$.visitorSupport[0].supportDetails").isEqualTo(updateRequest.supportList!![0].supportDetails!!)
-        .jsonPath("$.visitorConcerns").isEqualTo(updateRequest.visitorConcerns!!)
+        .jsonPath("$.visitNotes.length()").isEqualTo(updateRequest.visitNotes!!.size)
+        .jsonPath("$.visitNotes[0].type").isEqualTo(updateRequest.visitNotes!![0].type.name)
```
Comment (philip-milne) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:577]: Does this need to check the text has changed?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt (hunk 1 above)
Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:585]: Yep I suppose so,... even though it makes no difference to the type as its generic code... but I suppose that is the dev knowledge and I can't test like that!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt (hunk 2 above)
Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:577]: yes for sure... missed that for some reason

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt (hunk 1 above)
Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:589]: actually had a second look at it and I think it's okay as it as there is no code that uses what the type it is! will chant about it!

Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/resource/VisitResourceIntTest.kt:589]: This should be dealt with during amend storey VB-600
PUT /visit/{ref}/notes/

**Comment by philip-milne** [comment]
LGTM

**Review by philip-milne — APPROVED** [review]

**Pull request merged**